WebDriverAgentLib/Utilities/FBXCAXClientProxy.m

```diff
@@ -88,9 +89,12 @@ - (XCElementSnapshot *)snapshotForElement:(XCAccessibilityElement *)element
                                  maxDepth:(nullable NSNumber *)maxDepth
                                     error:(NSError **)error
 {
-  NSMutableDictionary *parameters = nil;
+  NSMutableDictionary *parameters = [[NSMutableDictionary alloc] init];
+  if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"13.0")) {
+    parameters[@"snapshotKeyHonorModalViews"] = @0;
```

Comment:
I think we have a setting for that value `FB_SETTING_INCLUDE_NON_MODAL_ELEMENTS`. Could we use it here instead of hardcoding it to zero?

Reply:
Also it would be more readable to use boolean `@(NO)`, which internally is also resolved to 0

Reply:
I've taken a deeper look at the XCTest framework, swizzled some functions to gain more insights and ran some tests to be sure - it seems that the behavior (at least with Xcode 12.5) is rather counter intuitive:
* Using the `includingNonModalElements` causes the same behavior as using the regular parent `XCUIElementQuery` - the `snapshotKeyHonorModalView` attribute is added to the snapshot request with a value of `0`.
* Using the opposite kind of sub-query - `excludingNonModalElements` - causes the `snapshotKeyHonorModalView` attribute to not be added at all (the snapshot request is sent without attributes at all), implying that testmanagerd's default behavior is actually to _exclude_ the non-modal views, unless being told otherwise (and telling it otherwise is the default behavior of the framework).

So it actually seems that the `includingNonModalElements` option doesn't make a difference when it comes to which elements are being retrieved, it's the opposite `excludingNonModalElements` that causes a different behavior, which is why at this point I believe we should keep this option hardcoded unless we refactor the code to use the excluding option instead.

Reply:
I've taken a deeper look at the XCTest framework, swizzled some functions to gain more insights and ran some tests to be sure - it seems that the behavior (at least with Xcode 12.5) is rather counter intuitive:

* Using the includingNonModalElements causes the same behavior as using the regular parent XCUIElementQuery - the snapshotKeyHonorModalView attribute is added to the snapshot request with a value of 0.
* Using the opposite kind of sub-query - excludingNonModalElements - causes the snapshotKeyHonorModalView attribute to not be added at all (the snapshot request is sent without attributes at all), implying that testmanagerd's default behavior is actually to exclude the non-modal views, unless being told otherwise (and telling it otherwise is the default behavior of the framework).

So it actually seems that the includingNonModalElements option doesn't make a difference when it comes to which elements are being retrieved, it's the opposite excludingNonModalElements that causes a different behavior, which is why at this point I believe we should keep this option hardcoded unless we refactor the code to use the excluding option instead.

Reply:
could you please add the below explanation there as a comment? It might help to understand that code in the future